poseidon2/src/external.rs

```diff
@@ -145,15 +145,38 @@ pub fn mds_light_permutation<
     }
 }
 
-/// A struct which holds the constants for the external layer.
+/// Constants used in the external layers of the Poseidon2 permutation.
+///
+/// External layers are applied at the beginning and end of the permutation.
+/// They help strengthen diffusion and prevent structural or algebraic attacks.
+///
+/// This struct stores round-specific constants for both initial and terminal external layers.
+/// These constants are combined with nonlinear S-boxes and lightweight matrix operations,
+/// improving security without adding much overhead.
+///
+/// Main purposes of these constants:
+/// - Inject randomness between rounds.
+/// - Support the Hades strategy (full nonlinear rounds).
+/// - Work with efficient linear layers in Poseidon2.
```

Comment:
This is probably a little too much info about the External Linear layer here. We could add an overview of it to the file comment?
```suggestion
/// A struct which stores round-specific constants for both initial and terminal external layers.
```